client/state/sites/domains/selectors.js

```diff
@@ -17,6 +21,26 @@ export const getDomainsBySiteId = ( state, siteId ) => {
 	return state.sites.domains.items[ siteId ] || EMPTY_SITE_DOMAINS;
 };
 
+/**
+ * Returns the  wpcom domain for the proved site id.
+ *
+ * @param {object} state - global state tree
+ * @param {number} siteId - identifier of the site
+ * @returns {object} the wpcom domain
+ */
+export const getWpComDomainBySiteId = ( state, siteId ) => {
+	const domains = getDomainsBySiteId( state, siteId );
+	if ( ! Array.isArray( domains ) ) {
+		return {};
```

Comment:
It's so weird to return an empty object. Those are harder to reason about and harder to check if they're empty. Is there a reason why we don't introduce a `null` or `undefined` in that scenario?

Reply:
Well since these are selectors, they are functions that would be used inside async data flows where sometimes there is a possibility for data to be empty.
For example the domains need to be async loaded and before that they will be null. If we send null if the data is not loaded and if someone destructures this object, their code will crash.
By sending an object when the data is not available we can allow the consumer of this object to safely destructure the object and just check whether the value he is looking for is undefined if required.

By sending in a null when no value is found we force the user to do a null check on the entire output.

This was my reasoning behind this approach :)
But I know it looks a bit dirty at the first glance. Will try to use null with optional chaining here 👍